src/Dashboard.Application/Services/ProjectService.cs

```diff
@@ -153,5 +153,19 @@ public Task<IEnumerable<Project>> GetAllProjectsAsync()
             
             await _projectRepository.SaveAsync();
         }
+
+        private async Task<int> GetNumberOfDiscoveryPipelines(int projectId)
+        {
+            var dynamicPanels = await _panelRepository.FindAllAsync(p => p.Discriminator == "DynamicPipelinesPanel" && p.ProjectId == projectId);
+            if (dynamicPanels == null)
+                return 0;
+            int dynamicPipelines = 0;
+            foreach (var panel in dynamicPanels)
+            {
+                dynamicPipelines += ((DynamicPipelinesPanel)panel).HowManyLastPipelinesToRead;

```

Comment:
Jak zrobisz to nowe repository DynamicPipelinesPanelsRepository, to bedziesz mogl przerzucic tego foreach na SQLa w bazie xd